node/external/transactionAPI/gasUsedAndFeeProcessor.go

```diff
@@ -58,45 +59,31 @@ func (gfp *gasUsedAndFeeProcessor) computeAndAttachGasUsedAndFee(tx *transaction
 	}
 
 	hasRefundForSender := false
-	totalRefunds := big.NewInt(0)
 	for _, scr := range tx.SmartContractResults {
 		if !scr.IsRefund || scr.RcvAddr != tx.Sender {
 			continue
 		}
 
+		gfp.setGasUsedAndFeeBaseOnRefundValue(tx, userTx, scr.Value)
 		hasRefundForSender = true
-		totalRefunds.Add(totalRefunds, scr.Value)
+		break
```

Comment:
I think we still need to iterate over all possible refunds (in the case of SC calls with multiple refunds for example), accumulate them and then set the total value.